### ISSUE TITLE
NPCReplacement doesn't work

### ISSUE THREAD

**Issue by WildBeef** [thommcgrath/Beacon#247]
I've been trying to get this code to work with no luck. It will remove the first creature just fine, but will not spawn the replacement. What am I doing wrong? I have it under the [/script/shootergame.shootergamemode] section all in one line.

`NPCReplacements=(FromClassName="Dodo_Character_BP_C",ToClassName="Jerboa_Character_BP_C")
`
I have experimented with several variations by moving parentheses and quotes around, but either end up with an error in the log files or it still just removed the first class ID. It's like it doesn't make it to the second part at all.

Any help with this would be much appreciated,

**Comment by thommcgrath** [comment]
It's an Ark issue really. From the guide at https://usebeacon.app/help/creature_adjustments_editor:

> Be aware that users have reported mixed success with this option however. In theory, it should be possible to replace all Dodos with Gigas to completely ruin a server. In practice, it appears the replacement creatures are still bound by their spawn point rules and limits. This means replacing a creature with one that does not normally spawn on the map may not spawn either creature at all.

In short, if the desired creature isn't already loaded into memory, you might as well have written `NPCReplacements=(FromClassName="Dodo_Character_BP_C",ToClassName="AlienHoobajoob")` because Ark doesn't know what `Jerboa_Character_BP_C` is.

Luckily, Beacon has a painless solution. In the "Tools" item at the top of your project, choose "Convert Creature Replacements to Spawn Point Additions" and it'll make the required changes to your project. It'll disable the dodo from spawning, and add jerboas to any spawn point where dodos would spawn, taking care to match frequency and limits.

**Comment by WildBeef** [comment]
Thanks for the fast reply!

That is weird. How does it differ some simply spawning a jerboa in with a summon command? I'm not the most knowledgeable on that backend stuff. 

The main reason I was trying to use this method is that it has an option to be used in the new dynamic config, which also wouldn't work when I tried(the file does just not that code written in its slightly new way). Which makes more sense now if it's not an error in the code.  Is there any way to use the add and subtract options in the dynamic file? If not I'll just do what you suggested and deal with server shutdowns. 

Is there any way to add them to the memory? Like a code that adds all dino spawns to an area with 0 weight or something so it's loaded and would then work in other areas? 

Either way thanks again for clearing that up! Been messing with that code for days now, it was driving me mad lol. 

Cheers!

**Comment by thommcgrath** [comment]
No, the dynamic config only allows a handful of configs. Your idea about adding them to a spawn at 0 weight is worth a shot. I've never tried it, nor have I heard of anybody else trying it. The full guide for the spawn editor is available at https://usebeacon.app/help/creature_spawns_editor. I would make sure both the spawn set weight and limit is set to 0.

**Comment by WildBeef** [comment]
Well, the idea didn't work. However is still seems like the code itself just doesn't work. 

It seemed like a good idea to see if my idea would work with them actively spawning somewhere else before trying with no weight, so I added Mantis so the Archaeopteryx spawn entry at a weight large enough to ensure it spawned:


`ConfigAddNPCSpawnEntriesContainer=(NPCSpawnEntriesContainerClassString="DinoSpawnEntriesArcha_C",NPCSpawnEntries=((AnEntryName="Mantis Memory",EntryWeight=20,NPCsToSpawnStrings=("Mantis_Character_BP_C"))),NPCSpawnLimits=((NPCClassString="Mantis_Character_BP_C",MaxPercentageOfDesiredNumToAllow=100)))
`
After a dino wipe I checked the redwoods and found one pretty fast. I ran a GetAllState on them just to be sure and shut back down. The log showed 30 or so Mantis so that was working. So I then added an NPCReplacment code to swap dodos for mantis:

`NPCReplacements=(FromClassName="Dodo_Character_BP_C",ToClassName="Mantis_Character_BP_C")
`

After a dino wipe and sitting in slomo 100 for a minutes, I searched the beaches to find no dodos or mantis still.
I then rebooted and added another NPCReplacement code to swap Parasaurs for Direwolves, to try it with something already on the map:

`NPCReplacements=(FromClassName="Para_Character_BP_C",ToClassName="Direwolf_Character_BP_C")`

I got back on, dinowiped, waited a bit, and went looking on the beaches. No dodos, No mantis, No paras, and no wolves. I rebooted once more and added in another code to replace trikes with dodos, since they are often found in the same spawn entry. 

`NPCReplacements=(FromClassName="Trike_Character_BP_C",ToClassName="Dodo_Character_BP_C")
`
When I looked around, it was the same was before but now theres no trikes. I did a GetAllState on dodos, paras, and trikes. It found many tekparas but that is it. 

All of these were put in the Game.ini under the _[/script/shootergame.shootergamemode]_ section. If theres an order of operations that matters at all, they were in the order listed. 

So at this point it seems like it just doesn't work at all? Wouldn't the last two have worked fine being normal dinos to spawn on that map?

One other thing I found odd is that during all my previous tests with changing the code around to see if it just wasn't making it to the second class name, I sometimes got an error that read

`]Warning: Unknown property in ClassNameReplacement:`

I have never seen ClassNameReplacement get used anywhere in any of these codes. Not sure if thats relevant or not. 

Any idea why it isn't even replacing stuff thats already on the island?

Thanks again!

**Comment by thommcgrath** [comment]
Hmm. Yeah at least some of those should have worked. I don't have much more I could add right now, but based on these results, I think you'll be better off making the changes with the spawn point editing.
